src/eterna/pose2D/Pose2D.ts

```diff
@@ -317,39 +349,61 @@ export default class Pose2D extends ContainerObject implements Updatable {
         return this._zoomLevel;
     }
 
+    /**
+     * @param animate should the zoom happen in an animation over time, instead of instantaniously
+     * @param center should the camera move to the center of the screen
+     */
     public setZoomLevel(zoomLevel: number, animate: boolean = true, center: boolean = false): void {
-        if ((this._zoomLevel !== zoomLevel || center) && animate) {
-            if (this._zoomLevel === zoomLevel && center) {
-                if (Math.abs(this._width / 2 - this._offX) + Math.abs(this._height / 2 - this._offY) < 50) {
+        const needsToZoom = this._zoomLevel !== zoomLevel;
+        if ((needsToZoom || center) && animate) {
+            if (!needsToZoom && center) {
+                // TODO Guy: hypot is expensive and I don't even need the sqrt
+                const distanceFromCenter = Math.hypot(
+                    this._width / 2 - this._offset.x,
+                    this._height / 2 - this._offset.y
+                );
+                if (distanceFromCenter < 50) {
                     return;
                 }
             }
 
-            this._startOffsetX = this._offX;
-            this._startOffsetY = this._offY;
-
-            let scaler = 1;
-            if (zoomLevel > this._zoomLevel) {
-                scaler = Pose2D.ZOOM_SPACINGS[zoomLevel] / Pose2D.ZOOM_SPACINGS[this._zoomLevel];
-            }
+            this._startOffset = this._offset.clone();
 
-            if (!this._offsetTranslating && !center) {
-                this._endOffsetX = scaler * (this._offX - this._width / 2) + this._width / 2;
-                this._endOffsetY = scaler * (this._offY - this._height / 2) + this._height / 2;
-            } else if (this._offsetTranslating) {
-                this._endOffsetX = scaler * (this._endOffsetX - this._width / 2) + this._width / 2;
-                this._endOffsetY = scaler * (this._endOffsetY - this._height / 2) + this._height / 2;
+            if (center) {
+                this._endOffset = new Point(this._width / 2, this._height / 2);
             } else {
-                this._endOffsetX = this._width / 2;
-                this._endOffsetY = this._height / 2;
+                // TODO: should this be different? I need to respect if the mosue moved
```

Comment:
Alright, I finally found it:

https://user-images.githubusercontent.com/6932278/153938950-77160248-3d7d-4f5d-8316-8c9ae38f251d.mp4

What happens is that if I zoom in, and _while the animation is playing_, move my mouse and then zoom in again, the mouse will not stay in the same world position (in this case it started outside the RNA, just to its right, but finished on the RNA)

I couldn't find a trivial and clean way to do this (which would happen by giving the screen/world conversion functions the current spacing in the middle of the animation, which isn't part of the ZOOM_SPACING array, but rather a linear value between the two).
I could calculate the value there, but that would mean that we'd calculate the progression in 3 different places, which I think is overkill.

Should I work on fixing that? Or should I document it as a known issue?

Reply:
Actually, I'm not sure if that would be enough to fix it, since the animation has to finish before we can zoom in again (or before the next zoom-in applies, IDK), so the mouse would have to move in world space either way

Reply:
I'll open a new issue and resolve this thread before merging